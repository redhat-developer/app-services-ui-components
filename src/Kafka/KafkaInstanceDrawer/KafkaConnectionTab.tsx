import {
  Button,
  ButtonVariant,
  ClipboardCopy,
  ExpandableSection,
  Label,
  Popover,
  Skeleton,
  Text,
  TextContent,
  TextVariants,
} from "@patternfly/react-core";
import { HelpIcon } from "@patternfly/react-icons";
import { useState } from "react";
import type { FunctionComponent } from "react";
import { Trans, useTranslation } from "react-i18next";
import { Link } from "react-router-dom";

export type KafkaConnectionTabProps = {
  isKafkaPending?: boolean;
  externalServer?: string;
  tokenEndPointUrl: string;
  linkToServiceAccount: string;
  linkToAccessTab: string;
  adminAPIUrl: string | undefined;
  showCreateServiceAccountModal: () => void;
  kafkaFleetManagerUrl: string;
};

export const KafkaConnectionTab: FunctionComponent<KafkaConnectionTabProps> = ({
  isKafkaPending,
  externalServer,
  tokenEndPointUrl,
  linkToServiceAccount,
  linkToAccessTab,
  adminAPIUrl,
  kafkaFleetManagerUrl,
  showCreateServiceAccountModal,
}) => {
  const { t } = useTranslation();

  const [isExpanded, setIsExpanded] = useState<boolean>(false);

  const onChangeExpandedSection = () => {
    setIsExpanded(!isExpanded);
  };

  return (
    <div className="mas--details__drawer--tab-content">
      <TextContent className="pf-u-pb-sm">
        <Text component={TextVariants.small}>
          {t("kafka:connection_tab.drawer_resource_tab_body_description_1")}
        </Text>
        <Text component={TextVariants.h3} className="pf-u-mt-xl">
          {t("kafka:connection_tab.bootstrap_server")}
        </Text>
        <Text component={TextVariants.small}>
          {t("kafka:connection_tab.bootstrap_server_description")}
        </Text>
        {isKafkaPending ? (
          <Skeleton fontSize="2xl" />
        ) : (
          <ClipboardCopy
            data-testid="drawerStreams-copyBootstrapURL"
            textAriaLabel={t("kafka:connection_tab.bootstrap_server")}
            isReadOnly
          >
            {externalServer}
          </ClipboardCopy>
        )}
      </TextContent>
      <TextContent className="pf-u-pb-sm">
        <Text component={TextVariants.h3} className="pf-u-mt-xl">
          {t("kafka:connection_tab.service_accounts_small")}
        </Text>
        <Text component={TextVariants.small}>
          {
            <Trans
              i18nKey={
                "kafka:connection_tab.create_service_account_to_generate_credentials"
              }
              components={{
                value: (
                  <Link
                    to={linkToServiceAccount}
                    data-testid="tableStreams-linkKafka"
                  ></Link>
                ),
              }}
            />
          }
        </Text>
      </TextContent>
      <Button
        variant={ButtonVariant.secondary}
        isInline
        onClick={showCreateServiceAccountModal}
        data-testid="drawerStreams-buttonCreateServiceAccount"
      >
        {t("kafka:connection_tab.create_service_account")}
      </Button>
      <TextContent className="pf-u-pt-sm">
        <Text component={TextVariants.small}>
          {
            <Trans
              i18nKey={"kafka:connection_tab.current_instance"}
              components={{
                value: <Link to={linkToAccessTab}></Link>,
              }}
            />
          }
        </Text>
      </TextContent>
      <ExpandableSection
        toggleContent={
          <div className="pf-c-content">
            <Text component={TextVariants.h3}>
              {t("kafka:connection_tab.rest_api_header")}
            </Text>
          </div>
        }
        isExpanded={isExpanded}
        onToggle={onChangeExpandedSection}
        className="pf-u-mt-lg"
      >
        <TextContent className="pf-u-pb-sm">
          <Text component={TextVariants.small} className={"pf-c-content"}>
            {t("kafka:connection_tab.rest_api_description")}
          </Text>
        </TextContent>
        <TextContent>
          <strong>
            {t("kafka:connection_tab.kafka_instance_url_label")}
            <Popover
              headerContent={
                <div>
                  {t("kafka:connection_tab.kafka_instance_url_popover_label")}
                </div>
              }
              bodyContent={
                <div>
                  {t("kafka:connection_tab.kafka_instance_url_popover_body")}
                </div>
              }
            >
              <Button
                variant={ButtonVariant.plain}
                aria-label={t(
                  "kafka:connection_tab.kafka_instance_url_button_aria_label"
                )}
              >
                <HelpIcon />
              </Button>
            </Popover>
          </strong>
          {adminAPIUrl ? (
            <ClipboardCopy
              textAriaLabel={t("kafka:connection_tab.kafka_instance_url_label")}
              isReadOnly
            >
              {adminAPIUrl}
            </ClipboardCopy>
          ) : (
            <Skeleton fontSize="2xl" />
          )}
          <TextContent className="pf-u-pt-sm">
            <Text component={TextVariants.small}>
              {t("kafka:connection_tab.kafka_instance_url_description")}
            </Text>
          </TextContent>
        </TextContent>
        <TextContent className="pf-u-mt-md">
          <strong>
            {t("kafka:connection_tab.kafka_management_url")}
            <Popover
              headerContent={
                <div>
                  {t("kafka:connection_tab.kafka_management_popover_header")}
                </div>
              }
              bodyContent={
                <div>
                  {t("kafka:connection_tab.kafka_management_popover_body")}
                </div>
              }
            >
              <Button
                variant={ButtonVariant.plain}
                aria-label={t(
                  "kafka:connection_tab.kafka_management_button_aria_label"
                )}
              >
                <HelpIcon />
              </Button>
            </Popover>
          </strong>
          {isKafkaPending ? (
            <Skeleton fontSize="2xl" />
          ) : (
            <ClipboardCopy
              textAriaLabel={t("kafka:connection_tab.kafka_management_url")}
              isReadOnly
            >
              {kafkaFleetManagerUrl}
            </ClipboardCopy>
          )}
          <TextContent className="pf-u-pt-sm">
            <Text component={TextVariants.small}>
              {t("kafka:connection_tab.kafka_management_description")}
            </Text>
          </TextContent>
        </TextContent>
      </ExpandableSection>
      <TextContent className="pf-u-pb-sm">
        <Text component={TextVariants.h3} className="pf-u-mt-xl">
          {t("kafka:connection_tab.authentication_method")}
        </Text>
        <Text component={TextVariants.h4} className="pf-u-mt-md">
          {t("kafka:connection_tab.sasl_oauthbearer")}{" "}
          <Label color="green">{t("kafka:connection_tab.recommended")}</Label>
          <Popover
            aria-label={t("kafka:connection_tab.sasl_oauthbearer")}
            bodyContent={
              <div>
                {t("kafka:connection_tab.sasl_oauthbearer_popover_content")}
              </div>
            }
          >
            <Button
              variant={ButtonVariant.plain}
              aria-label={t(
                "kafka:connection_tab.more_info_about_sasl_oauthbearer"
              )}
            >
              <HelpIcon />
            </Button>
          </Popover>
        </Text>
        <Text component={TextVariants.small}>
          {t("kafka:connection_tab.sasl_oauthbearer_description")}
        </Text>
        <strong>{t("kafka:connection_tab.token_endpoint_url")}</strong>
        {isKafkaPending ? (
          <Skeleton fontSize="2xl" />
        ) : (
          <ClipboardCopy
            textAriaLabel={t("kafka:connection_tab.token_endpoint_url")}
            isReadOnly
          >
            {tokenEndPointUrl}
          </ClipboardCopy>
        )}
      </TextContent>
      <TextContent className="pf-u-pb-sm">
        <Text component={TextVariants.h4} className="pf-u-mt-md">
          {t("kafka:connection_tab.sasl_plain")}
        </Text>
        <Text component={TextVariants.small}>
          {t("kafka:connection_tab.sasl_plain_description")}
        </Text>
      </TextContent>
    </div>
  );
};
