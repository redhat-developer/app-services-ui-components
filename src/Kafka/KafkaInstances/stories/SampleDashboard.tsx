export default () => <div>Dashboard</div>;
