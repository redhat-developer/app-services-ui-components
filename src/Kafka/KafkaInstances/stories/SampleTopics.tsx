export default () => <div>Topics</div>;
