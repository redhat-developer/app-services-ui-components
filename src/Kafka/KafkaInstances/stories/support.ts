import { addHours } from "date-fns";
import { useEffect, useState } from "react";
import type { KafkaInstance } from "../../types";

const now = new Date().toISOString();
const future = addHours(new Date(), 19).toISOString();
const INSTANCES: KafkaInstance[] = [
  {
    id: "1",
    name: "foo",
    createdAt: now,
    updatedAt: now,
    expiryDate: undefined,
    owner: "owner",
    provider: "aws",
    region: "region",
    status: "provisioning",
    plan: "standard",
    size: "1",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
  {
    id: "2",
    name: "bar",
    createdAt: now,
    updatedAt: now,
    expiryDate: future,
    owner: "owner",
    provider: "aws",
    region: "region",
    status: "deleting",
    plan: "developer",
    size: "1",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
  {
    id: "3",
    name: "baz",
    createdAt: now,
    updatedAt: now,
    expiryDate: future,
    owner: "owner",
    provider: "azure",
    region: "region",
    status: "ready",
    plan: "standard",
    size: "2",
    ingress: undefined,
    egress: undefined,
    storage: undefined,
    maxPartitions: undefined,
    connections: undefined,
    connectionRate: undefined,
    messageSize: undefined,
    billing: undefined,
  },
];
export const useInstances = () => {
  const [instances, setInstances] = useState<KafkaInstance[]>();

  useEffect(() => {
    const timer = setTimeout(() => {
      setInstances(INSTANCES);
    }, 500);
    return () => clearTimeout(timer);
  }, []);
  return {
    instances,
    loading: instances === undefined,
  };
};
export const useInstance = (id: string) => {
  const { instances, loading } = useInstances();
  const instance = instances?.find((i) => i.id === id);

  return {
    instance,
    loading: loading || instance === undefined,
  };
};
