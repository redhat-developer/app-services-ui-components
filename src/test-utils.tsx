import type { RenderOptions, RenderResult } from "@testing-library/react";
import { act, render, waitFor } from "@testing-library/react";
import type { FunctionComponent, ReactElement } from "react";
import { Suspense } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { I18nProvider } from "./I18n";
import { AlertProvider } from "../src/shared/Alerts";

const suspenseTestId = "i18n-suspense";

const AllTheProviders: FunctionComponent = ({ children }) => {
  return (
    <Router>
      <AlertProvider>
        <I18nProvider
          lng={"en"}
          resources={{
            en: {
              common: () => import("../locales/en/common.json"),
              "create-kafka-instance": () =>
                import("../locales/en/create-kafka-instance.json"),
              kafka: () => import("../locales/en/kafka.json"),
              metrics: () => import("../locales/en/metrics.json"),
              overview: () => import("../locales/en/overview.json"),
              datascienceoverview: () =>
                import("../locales/en/datascienceoverview.json"),
              kafkaoverview: () => import("../locales/en/kafkaoverview.json"),
              topic: () => import("../locales/en/topic.json"),
              apimgmtoverview: () =>
                import("../locales/en/apimgmtoverview.json"),
              "manage-kafka-permissions": () =>
                import("../locales/en/manage-kafka-permissions.json"),
              "create-topic": () => import("../locales/en/create-topic.json"),
              "message-browser": () =>
                import("../locales/en/message-browser.json"),
              "overview-v2": () => import("../locales/en/overview-v2.json"),
              "overview-v3": () => import("../locales/en/overview-v3.json"),
              "overview-v4": () => import("../locales/en/overview-v4.json"),
              "kafkaoverview-v2": () =>
                import("../locales/en/kafkaoverview-v2.json"),
              "kafkaoverview-v3": () =>
                import("../locales/en/kafkaoverview-v3.json"),
              "connection-tab": () =>
                import("../locales/en/connection-tab.json"),
              "service-registry": () =>
                import("../locales/en/service-registry.json"),
              "connection-tab-p1": () =>
                import("../locales/en/connection-tab-p1.json"),
            },
            it: {
              common: () => Promise.resolve({ delete: "Elimina" }),
            },
          }}
          debug={false}
        >
          <Suspense fallback={<div data-testid={suspenseTestId}>loading</div>}>
            {children}
          </Suspense>
        </I18nProvider>
      </AlertProvider>
    </Router>
  );
};

const AllTheProvidersWithRoot: FunctionComponent = ({ children }) => (
  <AllTheProviders>
    <div id={"root"}>{children}</div>
  </AllTheProviders>
);
const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const renderDialog = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProvidersWithRoot, ...options });

async function waitForI18n(r: RenderResult) {
  await waitFor(() => {
    expect(r.queryByTestId(suspenseTestId)).not.toBeInTheDocument();
  });
}

async function waitForPopper() {
  await act(async () => {
    /* let popper do its updates */
  });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderDialog, waitForI18n, waitForPopper };
