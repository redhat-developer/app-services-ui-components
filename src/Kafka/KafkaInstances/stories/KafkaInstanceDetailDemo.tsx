import {
  Breadcrumb,
  BreadcrumbItem,
  Button,
  Level,
  LevelItem,
  PageSection,
  Skeleton,
  Tab,
  TabContent,
  Tabs,
  TabsComponent,
  Title,
} from "@patternfly/react-core";
import type { ComponentType } from "react";
import { lazy, Suspense, useRef } from "react";
import {
  Link,
  Redirect,
  Route,
  Switch,
  useHistory,
  useParams,
  useRouteMatch,
} from "react-router-dom";
import { Loading } from "../../../shared";
import { useKafkaInstanceDrawer } from "../KafkaInstanceDrawer";
import { useInstance } from "./support";

const LazyDashboard = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleDashboard")), 1500)
    )
);

const LazyTopics = lazy(
  () =>
    new Promise<{ default: ComponentType<any> }>((resolve) =>
      setTimeout(() => resolve(import("./SampleTopics")), 1500)
    )
);

export const KafkaInstanceDetailDemo = () => {
  const { id, tab = "dashboard" } = useParams<{
    id: string;
    tab: string | undefined;
  }>();
  const history = useHistory();
  const match = useRouteMatch({ path: "/streams/:id" });
  const { instance } = useInstance(id);
  const { openDrawer } = useKafkaInstanceDrawer();
  const dashboardRef = useRef<HTMLDivElement>(null);
  const topicsRef = useRef<HTMLDivElement>(null);
  return (
    <>
      <PageSection variant={"light"} type={"breadcrumb"}>
        <Breadcrumb>
          <BreadcrumbItem
            render={() => <Link to={"/streams"}>Kafka instances</Link>}
          />
          <BreadcrumbItem isActive>
            {instance?.name || <Skeleton width={"150px"} />}
          </BreadcrumbItem>
        </Breadcrumb>
      </PageSection>
      <PageSection variant={"light"}>
        <Level>
          <LevelItem>
            <Title headingLevel={"h1"}>
              {instance?.name || <Skeleton width={"150px"} />}
            </Title>
          </LevelItem>
          <LevelItem>
            {instance && (
              <Button onClick={() => openDrawer(instance)}>Open drawer</Button>
            )}
          </LevelItem>
        </Level>
      </PageSection>
      <PageSection variant={"light"} type={"tabs"}>
        <Tabs
          activeKey={tab}
          onSelect={(_, key) => {
            history.push({
              pathname: `${match?.url || ""}/${key}`,
            });
          }}
          component={TabsComponent.nav}
          aria-label="Tabs in the toggled separate content example"
        >
          <Tab
            eventKey={"dashboard"}
            title="Dashboard"
            tabContentId="dashboard-tab"
          />
          <Tab eventKey={"topics"} title="Topics" tabContentId="topics-tab" />
        </Tabs>
      </PageSection>
      <PageSection isFilled={true}>
        <Suspense fallback={<Loading />}>
          <Switch>
            <Route path={`${match?.path || ""}/dashboard`}>
              <TabContent
                eventKey={"dashboard"}
                id="dashboard-tab"
                ref={dashboardRef}
                aria-label="TODO"
              >
                <LazyDashboard />
              </TabContent>
            </Route>
            <Route path={`${match?.path || ""}/topics`}>
              <TabContent
                eventKey={"topics"}
                id="topics-tab"
                ref={topicsRef}
                aria-label="TODO"
              >
                <LazyTopics />
              </TabContent>
            </Route>
            <Redirect
              path={match?.path}
              exact={true}
              to={`${match?.path || ""}/dashboard`}
            />
          </Switch>
        </Suspense>
      </PageSection>
    </>
  );
};

export default KafkaInstanceDetailDemo;
