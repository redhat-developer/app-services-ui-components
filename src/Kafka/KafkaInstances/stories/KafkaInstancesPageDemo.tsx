import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
