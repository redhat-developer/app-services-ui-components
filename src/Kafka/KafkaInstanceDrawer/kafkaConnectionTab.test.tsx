import { composeStories } from "@storybook/testing-react";
import { render, waitForI18n } from "../../test-utils";
import * as stories from "./KafkaConnectionTab.stories";

const { ConnectionTab, ConnectionTabWhenkafkaCreationPending } =
  composeStories(stories);

describe("ConnectionTab", () => {
  it("renders", async () => {
    const comp = render(<ConnectionTab />);
    await waitForI18n(comp);
    expect(comp.baseElement).toMatchSnapshot();
  });

  it("should render bootstrap server and Token endpoint URL value when kafka instance creation is done", async () => {
    const comp = render(<ConnectionTab />);
    await waitForI18n(comp);
    expect(await comp.findByLabelText("Bootstrap server")).toHaveDisplayValue(
      "hema-test-c-k-l-kafka-stage.rhcloud.com:443"
    );
    expect(await comp.findByLabelText("Token endpoint URL")).toHaveDisplayValue(
      "https://identify.api.stage.openshift.com/auth/realms/rhoas/protocol/openid-connect/token"
    );
    expect(
      await comp.findByLabelText("URL for Kafka Instance API")
    ).toHaveDisplayValue(
      "https://admin-server-[name of instance]-[domain]/openapi"
    );
    expect(
      await comp.findByLabelText("URL for Kafka Management API")
    ).toHaveDisplayValue(
      "https://api.openshift.com/api/kafkas_mgmt/v1/openapi"
    );

    expect(
      comp.getByRole("button", { name: "Create service account" })
    ).toHaveAttribute(
      "data-testid",
      "drawerStreams-buttonCreateServiceAccount"
    );

    expect(
      comp.getByRole("link", { name: "Service Accounts" })
    ).toHaveAttribute("data-testid", "tableStreams-linkKafka");
  });

  it("bootstrap server and Token endpoint URL should be disabled when the kafka creating is pending", async () => {
    const comp = render(<ConnectionTabWhenkafkaCreationPending />);
    await waitForI18n(comp);
    expect(comp.queryByLabelText("Bootstrap server")).toBe(null);
    expect(comp.queryByLabelText("Token endpoint URL")).toBe(null);
    expect(comp.queryByLabelText("URL for Kafka Instance API")).toBe(null);
    expect(comp.queryByLabelText("URL for Kafka Management API")).toBe(null);
  });
});
