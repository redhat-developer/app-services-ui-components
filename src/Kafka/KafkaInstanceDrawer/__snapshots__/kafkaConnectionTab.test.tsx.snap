// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionTab renders 1`] = `
<body>
  <div>
    <div
      class="mas--details__drawer--tab-content"
    >
      <div
        class="pf-c-content pf-u-pb-sm"
      >
        <small
          class=""
          data-ouia-component-id="OUIA-Generated-Text-1"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Your application or tool initially connects to the Kafka instance using the bootstrap server and authenticates with your service account credentials.
        </small>
        <h3
          class="pf-u-mt-lg"
          data-ouia-component-id="OUIA-Generated-Text-2"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Bootstrap server
        </h3>
        <small
          class=""
          data-ouia-component-id="OUIA-Generated-Text-3"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Establish the initial connection between your client and Kafka instance.
        </small>
        <div
          class="pf-c-clipboard-copy"
          data-testid="drawerStreams-copyBootstrapURL"
        >
          <div
            class="pf-c-clipboard-copy__group"
          >
            <input
              aria-invalid="false"
              aria-label="Bootstrap server"
              class="pf-c-form-control"
              data-ouia-component-id="OUIA-Generated-TextInputBase-1"
              data-ouia-component-type="PF4/TextInput"
              data-ouia-safe="true"
              id="text-input-0"
              type="text"
              value="hema-test-c-k-l-kafka-stage.rhcloud.com:443"
            />
            <button
              aria-disabled="false"
              aria-label="Copy to clipboard"
              aria-labelledby="copy-button-0 text-input-0"
              class="pf-c-button pf-m-control"
              data-ouia-component-id="OUIA-Generated-Button-control-1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              id="copy-button-0"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="pf-c-content pf-u-pb-sm"
      >
        <h3
          class="pf-u-mt-xl"
          data-ouia-component-id="OUIA-Generated-Text-4"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Service accounts
        </h3>
        <small
          class=""
          data-ouia-component-id="OUIA-Generated-Text-5"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Create a service account to generate credentials. Manage the service accounts you create on the 
          <a
            href="/to-service-account"
          >
            Service Accounts
          </a>
           page.
        </small>
      </div>
      <button
        aria-disabled="false"
        class="pf-c-button pf-m-secondary"
        data-ouia-component-id="OUIA-Generated-Button-secondary-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe="true"
        type="button"
      >
        Create service account
      </button>
      <div
        class="pf-c-content pf-u-pt-sm"
      >
        <small
          class=""
          data-ouia-component-id="OUIA-Generated-Text-6"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Accounts do not have access to this instance unless granted under the instanceâ€™s 
          <a
            href="/to-access-tab"
          >
            Access tab
          </a>
          . The owner, the organization administrator, or users with the Allow Alter permission on this instance can manage access.
        </small>
      </div>
      <div
        class="pf-c-content pf-u-pb-sm"
      >
        <h3
          class="pf-u-mt-xl"
          data-ouia-component-id="OUIA-Generated-Text-7"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Authentication method
        </h3>
        <h4
          class="pf-u-mt-md"
          data-ouia-component-id="OUIA-Generated-Text-8"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          SASL/OAUTHBEARER
           
          <span
            class="pf-c-label pf-m-green"
          >
            <span
              class="pf-c-label__content"
            >
              Recommended
            </span>
          </span>
          <button
            aria-disabled="false"
            aria-label="More information about SASL/OAUTHBEARER"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 1024 1024"
              width="1em"
            >
              <path
                d="M521.3,576 C627.5,576 713.7,502 713.7,413.7 C713.7,325.4 627.6,253.6 521.3,253.6 C366,253.6 334.5,337.7 329.2,407.2 C329.2,414.3 335.2,416 343.5,416 L445,416 C450.5,416 458,415.5 460.8,406.5 C460.8,362.6 582.9,357.1 582.9,413.6 C582.9,441.9 556.2,470.9 521.3,473 C486.4,475.1 447.3,479.8 447.3,521.7 L447.3,553.8 C447.3,570.8 456.1,576 472,576 C487.9,576 521.3,576 521.3,576 M575.3,751.3 L575.3,655.3 C575.313862,651.055109 573.620137,646.982962 570.6,644 C567.638831,640.947672 563.552355,639.247987 559.3,639.29884 L463.3,639.29884 C459.055109,639.286138 454.982962,640.979863 452,644 C448.947672,646.961169 447.247987,651.047645 447.29884,655.3 L447.29884,751.3 C447.286138,755.544891 448.979863,759.617038 452,762.6 C454.961169,765.652328 459.047645,767.352013 463.3,767.30116 L559.3,767.30116 C563.544891,767.313862 567.617038,765.620137 570.6,762.6 C573.659349,759.643612 575.360354,755.553963 575.3,751.3 M512,896 C300.2,896 128,723.9 128,512 C128,300.3 300.2,128 512,128 C723.8,128 896,300.2 896,512 C896,723.8 723.7,896 512,896 M512.1,0 C229.7,0 0,229.8 0,512 C0,794.2 229.8,1024 512.1,1024 C794.4,1024 1024,794.3 1024,512 C1024,229.7 794.4,0 512.1,0"
              />
            </svg>
          </button>
        </h4>
        <small
          class=""
          data-ouia-component-id="OUIA-Generated-Text-9"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Use your service account credentials and a token endpoint URL to authenticate your client with the Kafka instance.
        </small>
        <h6
          class="pf-u-mt-md"
          data-ouia-component-id="OUIA-Generated-Text-10"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Token endpoint URL
        </h6>
        <div
          class="pf-c-clipboard-copy"
        >
          <div
            class="pf-c-clipboard-copy__group"
          >
            <input
              aria-invalid="false"
              aria-label="Token endpoint URL"
              class="pf-c-form-control"
              data-ouia-component-id="OUIA-Generated-TextInputBase-2"
              data-ouia-component-type="PF4/TextInput"
              data-ouia-safe="true"
              id="text-input-1"
              readonly=""
              type="text"
              value="https://identify.api.stage.openshift.com/auth/realms/rhoas/protocol/openid-coonect/token"
            />
            <button
              aria-disabled="false"
              aria-label="Copy to clipboard"
              aria-labelledby="copy-button-1 text-input-1"
              class="pf-c-button pf-m-control"
              data-ouia-component-id="OUIA-Generated-Button-control-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              id="copy-button-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 448 512"
                width="1em"
              >
                <path
                  d="M320 448v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24V120c0-13.255 10.745-24 24-24h72v296c0 30.879 25.121 56 56 56h168zm0-344V0H152c-13.255 0-24 10.745-24 24v368c0 13.255 10.745 24 24 24h272c13.255 0 24-10.745 24-24V128H344c-13.2 0-24-10.8-24-24zm120.971-31.029L375.029 7.029A24 24 0 0 0 358.059 0H352v96h96v-6.059a24 24 0 0 0-7.029-16.97z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="pf-c-content pf-u-pb-sm"
      >
        <h4
          class="pf-u-mt-md"
          data-ouia-component-id="OUIA-Generated-Text-11"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          SASL/PLAIN
        </h4>
        <small
          class=""
          data-ouia-component-id="OUIA-Generated-Text-12"
          data-ouia-component-type="PF4/Text"
          data-ouia-safe="true"
          data-pf-content="true"
        >
          Use your service account credentials (Client ID and Client secret) as the user name and password to authenticate your client with the Kafka instance.
        </small>
      </div>
    </div>
  </div>
</body>
`;
